src/rgw/rgw_rest_s3.cc

```diff
@@ -2644,8 +2644,16 @@ int RGWCreateBucket_ObjStore_S3::get_params(optional_yield y)
 
 void RGWCreateBucket_ObjStore_S3::send_response()
 {
-  if (op_ret == -ERR_BUCKET_EXISTS)
-    op_ret = 0;
+  if (op_ret == -ERR_BUCKET_EXISTS) {
+    const auto eexist_override = s->cct->_conf.get_val<bool>("rgw_bucket_eexist_override");
```

Comment:
if we're using the mutex version of `get_val()` (which seems fine here), please remove `with_legacy: true` from the config option. otherwise we're using additional memory for `_conf->rgw_bucket_eexist_override` without using it

Reply:
I generally prefer the legacy versions, but it's hard to see a specific advantage in this case

Reply:
I don't see much benefit from saving the effective number of bytes in memory either--couldn't be more than 4?--but, it should be one or the other, so removed with_legacy (and see_also, which was unused)